cmd/sysl/cmd_diagram.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/anz-bank/sysl/pkg/cmdutils"
+	"github.com/anz-bank/sysl/pkg/mermaid/endpointanalysisdiagram"
+	"github.com/anz-bank/sysl/pkg/mermaid/integrationdiagram"
+	"github.com/anz-bank/sysl/pkg/mermaid/sequencediagram"
+	"github.com/anz-bank/sysl/pkg/sysl"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type diagramCmd struct {
+	cmdutils.DiagramCmd
+}
+
+func (p *diagramCmd) Name() string { return "diagram" }
+
+func (p *diagramCmd) MaxSyslModule() int { return 1 }
+
+func (p *diagramCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	cmd := app.Command(p.Name(), "Generate mermaid diagrams").Alias("md")
+	cmd.Flag("integrationdiagram", "To generate the integration"+
+		" diagram, specify the application name").Short('i').StringVar(&p.IntegrationValue)
+	cmd.Flag("sequencediagram", "To generate the sequence diagram, specify the "+
+		"application name and endpoint in this format app->endpoint").Short('s').StringVar(&p.SequenceValue)
+	cmd.Flag("endpointanalysis", "To generates the endpoint analysis diagram"+
```

Comment:
Can this be something else that doesn't require the `true` parameter?

Reply:
Is there a way to check which flags have been populated by the user? If that can be done then I guess I can remove this. I'll check on that